R/pkgdown.R

```diff
@@ -47,6 +47,93 @@ use_pkgdown <- function(config_file = "_pkgdown.yml", destdir = "docs") {
   invisible(TRUE)
 }
 
+# tidyverse pkgdown setup ------------------------------------------------------
+
+#' @details
+#' * `use_tidy_pkgdown()`: Implements the pkgdown setup favored by the
+#'   tidyverse team:
+#'   - [use_pkgdown()] does basic local setup
+#'   - [use_github_pages()] prepares to publish the pkgdown site from the
+#'     `github-pages` branch
+#'   - [`use_github_action("pkgdown")`][use_github_action()] configures a
+#'     GitHub Action to automatically build the pkgdown site and deploy it via
+#'     GitHub Pages
+#'   - The pkgdown website URL (if specified or inferred) is added to the
+#'     pkgdown configuration file and to the URL field of DESCRIPTION
+#'
+#' @param cname Optional, custom domain name. If the target repo belongs to the
+#'   tidyverse or r-lib GitHub organizations, this is formed automatically.
+#'
+#' @rdname tidyverse
+#' @export
+use_tidy_pkgdown <- function(cname = NULL) {
```

Comment:
Just occurred to me: should we say something about tidytemplate for sites tidyverse org? Or maybe it's not worth it, given adding a new tidyverse package is a relatively rare occurrence.

Reply:
I think it's possibly we should offer ourselves better templating for pkgdown config, but I definitely would not get into that here in this PR or, probably for v2.0.0.